pkg/build/admission/defaults/admission.go

```diff
@@ -78,24 +79,24 @@ func (b BuildDefaults) applyPodDefaults(pod *kapi.Pod) {
 	defaultResources := b.config.Resources
 	for i := range pod.Spec.Containers {
 		podEnv := &pod.Spec.Containers[i].Env
-		util.MergeTrustedEnvWithoutDuplicates(b.config.Env, podEnv, false)
+		util.MergeTrustedEnvWithoutDuplicates(util.CopyApiEnvVarToV1EnvVar(b.config.Env), podEnv, false)
 
 		if pod.Spec.Containers[i].Resources.Limits == nil {
-			pod.Spec.Containers[i].Resources.Limits = kapi.ResourceList{}
+			pod.Spec.Containers[i].Resources.Limits = v1.ResourceList{}
 		}
 		for name, value := range defaultResources.Limits {
-			if _, ok := pod.Spec.Containers[i].Resources.Limits[name]; !ok {
+			if _, ok := pod.Spec.Containers[i].Resources.Limits[v1.ResourceName(name)]; !ok {
 				glog.V(5).Infof("Setting default resource limit %s for pod %s/%s to %s", name, pod.Namespace, pod.Name, value)
-				pod.Spec.Containers[i].Resources.Limits[name] = value
+				pod.Spec.Containers[i].Resources.Limits[v1.ResourceName(name)] = value
 			}
 		}
 		if pod.Spec.Containers[i].Resources.Requests == nil {
-			pod.Spec.Containers[i].Resources.Requests = kapi.ResourceList{}
+			pod.Spec.Containers[i].Resources.Requests = v1.ResourceList{}
```

Comment:
Where are you getting your pod from.  There is defaulting on resource.Requests.  Internal would always have run it.  Does this path?